src/cryptography/x509/name.py

```diff
@@ -325,3 +359,92 @@ def __len__(self) -> int:
     def __repr__(self) -> str:
         rdns = ",".join(attr.rfc4514_string() for attr in self._attributes)
         return "<Name({})>".format(rdns)
+
+
+class _RFC4514NameParser:
+    _OID_RE = re.compile(r"(\d|[1-9]\d+)(\.\d|[1-9]\d+)+")
```

Comment:
This regex will match invalid OIDs -- do we care?

Reply:
Invalid how? The results are then immediately passed to the OID constructor which will reject them. Want me to add a test?

Reply:
OID constructor rejection is fine.